microservices/now_etls/now_etls.py

```diff
@@ -24,7 +24,10 @@ def etl_now_submission_data():
         user=DB_USER, password=DB_PASS, dbname=DB_NAME)
 
     click.echo('Beginning ETL')
-    NOW_submissions_ETL(connection)
+    try:
+        NOW_submissions_ETL(connection)
+    finally:
+        connection.close()
```

Comment:
I just learned you can do this using `with`, which guarantees that the connection is closed. 

Looks like it the psycopg2 connection class supports it, https://pynative.com/python-postgresql-transaction-management-using-commit-and-rollback/

Reply:
I'm using with in the actual ETL function with the passed connection. From the documentation I read it only said that with committed transaction if it completed, not that it would close the connection if it failed. 

http://initd.org/psycopg/docs/usage.html#with-statement

There's actually a specific warning in the documentation saying that it *doesn't* close the connection.